File: server/router/kubeinfo.go
package router

import (
"gin-vue-admin/api/v1"
"gin-vue-admin/middleware"
"github.com/gin-gonic/gin"
)

func InitKubeRouter(Router *gin.RouterGroup) {
	AppsRouter := Router.Group("kubeinfo").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		AppsRouter.POST("restartPod", v1.RestartPod)             // 获取应用pod列表
		AppsRouter.GET("getPodList", v1.GetPodList)             // 获取应用pod列表
		AppsRouter.GET("getDeployment", v1.GetDeployment)             // 获取应用Deployment信息
		AppsRouter.GET("getDeploymentList", v1.GetDeploymentList)             // 获取应用Deployment列表

	}
}

File: web/src/api/kubeinfo.js
import service from '@/utils/request'



 export const getPodList = (params) => {
     return service({
         url: "/kubeinfo/getPodList",
         method: 'get',
         params
     })
}
 export const restartPod = (params) => {
     return service({
         url: "/kubeinfo/restartPod",
         method: 'post',
         params
     })
}
 export const getDeploymentList = (params) => {
     return service({
         url: "/kubeinfo/getDeploymentList",
         method: 'get',
         params
     })
    }
 export const getDeployment = (params) => {
     return service({
         url: "/kubeinfo/getDeployment",
         method: 'get',
         params
     })
 }    